h2o-docs/src/product/security.rst

```diff
@@ -709,3 +709,127 @@ Example:
+- ``h2o_ssl_protocol`` (optional): The protocol name used during encrypted communication (supported by JVM). This defaults to TSLv1.2.
+- ``h2o_ssl_enabled_algorithms`` (optional): A comma separated list of enabled cipher algorithms. Include only those that are supported by JVM.
+
+This must be set for every node in the cluster. Every node needs to have access to both Java keystore and Java truststore containing appropriate keys and certificates.
+
+This feature should not be used together with the ``-useUDP`` flag, as we currently do not support UDP encryption through DTLS or any other protocol that might result in unencrypted data transfers.
+
+Keystore/Truststore Generation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Keystore/truststore creation and distribution are deployment specific and have to be handled by the end user.
+
+Basic keystore/truststore generation can be done using the keytool program, which ships with Java, documentation can be found `here <https://docs.oracle.com/javase/7/docs/technotes/tools/solaris/keytool.html>`__. Each node should have a key pair generated, and all public keys should be imported into a single truststore, which should be distributed to all the nodes.
+
+The simplest (though not recommended) way would be to call:
+
+::
+
+  keytool -genkeypair -keystore h2o-internal.jks -alias h2o-internal
+
+Then distribute the ``h2o-internal.jks`` file to all the nodes, and set it as both the keystore and truststore in ``ssl.config``. 
+
+A more secure way would be to:
+
+1. Run the same command on each node:
+  
+ ::
+
+  keytool -genkeypair -keystore h2o-internal.jks -alias h2o-internal
+
+2. Extract the certificate on each node:
+
+ ::
+
+  keytool -export -keystore h2o-internal.jks -alias signFiles -file node<number>.cer
+
+3. Distribute all of the above certificates to each node, and on each node create a truststore containing all of them (or put all certificates on one node, import to truststore and distribute that truststore to each node):
+
+ ::
+
+  keytool -importcert -file node<number>.cer -keystore truststore.jks -alias node<number>
+
+
+Performance
+~~~~~~~~~~~
+
+Turning on SSL may result in performance overhead for settings and algorithms that exchange data between nodes due to encryption/decryption time. Some algorithms (such as Deep Learning) might also slower because of this.
+
+Example benchmark on a 5 node cluster (6GB memory per node) working with a 5.8mln row dataset (580MB):
+
++------------+---------------------+------------------------+
+|            | Non SSL             | SSL                    |
++============+=====================+========================+
+| Parsing:   | 4.908s              | 5.304s                 |
++------------+---------------------+------------------------+
+| GLM model: | 01:39.446           | 01:49.634              |
++------------+---------------------+------------------------+
+| DL model:  | 11:53.54 (42epochs) | 28:06.738 (100 epochs) |
```

Comment:
For now, I've taken out the DL row. Can we merge this and then add the DL benchmark/performance information when it becomes available.